test/metric/dimension/container_insights_provider.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package dimension
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/computetype"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"log"
+)
+
+type ContainerInsightsDimensionProvider struct {
+	Provider
+}
+
+var _ IProvider = (*ContainerInsightsDimensionProvider)(nil)
+
+func (p *ContainerInsightsDimensionProvider) IsApplicable() bool {
+	if (p.Provider.env.ComputeType == computetype.ECS) {
```

Comment:
could simplify this to just return the result of `p.Provider.env.ComputeType == computetype.ECS` similar to the one below

Reply:
oh wow andrew missed this!

Reply:
:(((((((((((( fine I'll look harder next time